### PR TITLE
presentations endpoint

### PR DESCRIPTION
**Pull request by jpslav**

### CODE REVIEW

File: app/bindings/api/v0/bindings/presentation.rb [hunk 1]
```diff
@@ -0,0 +1,234 @@
+=begin
+#Assessment Network Interactions API
+
+#Records interactions with content distributed by the Assessment Network.  Requests to this API should include `application/json` in the `Accept` header.  The desired API version is specified in the request URL, e.g. `...a15k.org/v0/flags`.          While the API does support a default version, that version will change over          time and therefore should not be used in production code!  Some endpoints require an API key to be passed in the request header.  There are two          types of API keys: API tokens and API IDs.  An API token is used for more restricted          access.  Such tokens should not be shared with end users.  API IDs are used for less          restricted access and may be exposed to clients (e.g. through use in browser-side code).          Both keys are available to members through www.a15k.org. 
+
```
Comment (nathanstitt) [inline app/bindings/api/v0/bindings/presentation.rb:5]: what's up with the weird spacing here?  When I viewed the file by itself all this is on one line, maybe you somehow are using windows line breaks or something?

Comment (nathanstitt) [reply app/bindings/api/v0/bindings/presentation.rb:5]: oh nm, I see this is auto-generated

File: app/controllers/api/v0/swagger_models.rb [hunk 2]
```diff
@@ -101,6 +93,39 @@ module Api::V0::SwaggerModels
     end
   end
 
+  swagger_schema :Presentation do
+    key :required, [:content_uid, :user_uid, :presented_at]
+    property :content_uid do
+      key :type, :string
+      key :description, "The a15k ID of the content being presented."
+    end
```
Comment (nathanstitt) [inline app/controllers/api/v0/swagger_models.rb]: for fields that are uuid, I think we should set the format https://swagger.io/specification/#dataTypes
```
  key :format, :uuid
```

File: app/models/flag.rb [hunk 3]
```diff
@@ -46,6 +46,33 @@ def self.find(id)
...
+  def self.find_all(*ids)
+    ids = [ids].flatten
+    return [] if ids.empty?
+
+    datas = redis.mget(ids.map{|id| redis_key(id)})
+
+    if datas.length != ids.length
+      missing_ids = datas.map.with_index {|data, ii| data.nil? ? ids[ii] : nil}.compact
+      raise NotAllItemsFound, "Could not find Flags with IDs #{missing_ids}"
+    end
```
Comment (nathanstitt) [inline app/models/flag.rb]: what happens if `data` and `ids` are the same length but contain different values?

I know Dante used to convert them to `Set` and then compare them, but you can also just do: `datas.sort == ids.sort`

File: app/bindings/api/v0/bindings/presentation.rb (hunk 1 above)
Comment (jpslav) [reply app/bindings/api/v0/bindings/presentation.rb:5]: yeah, I added `\` in the markdown string so it renders nicely on swagger-ui... unfortunately I guess that means it renders terribly here.

File: app/models/flag.rb (hunk 3 above)
Comment (jpslav) [reply app/models/flag.rb]: totes! should be `if datas.compact.length != ids.length`

**Pull request merged**